wweval/R/supplement_analyses_and_figs.R

```diff
@@ -0,0 +1,728 @@
+    scenario = "status_quo",
+    forecast_date = this_forecast_date,
+    model_type = "ww",
+    location = this_location,
+    output_type = glue::glue("hosp_quantiles"),
+    file_extension = "tsv"
+  )
+
+  if (file.exists(fp_ww)) {
+    hosp_quantiles_ww <- readr::read_tsv(fp_ww)
+
+    fp_hosp <- get_filepath(
+      eval_output_subdir,
+      scenario = "no_wastewater",
+      forecast_date = this_forecast_date,
+      model_type = "hosp",
+      location = this_location,
+      output_type = glue::glue("quantiles"),
+      file_extension = "tsv"
+    )
+    hosp_quantiles_hosp <- readr::read_tsv(fp_hosp)
+
+
+    min_scores_date <- min(scores_single_loc_date$date)
+
+    quantiles <- hosp_quantiles_ww |>
+      dplyr::bind_rows(hosp_quantiles_hosp)
+
+    quantiles_wide <- quantiles |>
+      dplyr::filter(
+        quantile %in% c(0.025, 0.25, 0.5, 0.75, 0.975)
+      ) |>
+      tidyr::pivot_wider(
+        id_cols = c(
+          location, forecast_date, period, scenario,
+          date, t, eval_data, calib_data, model_type
+        ),
+        names_from = quantile,
+        values_from = value
+      )
+
+    quantiles_wide_forecast <- quantiles_wide |>
+      dplyr::filter(date >= min_scores_date)
+
+
+    scores_avg <- scores_single_loc_date |>
+      dplyr::group_by(forecast_date, location, model) |>
+      dplyr::summarize(avg_crps = mean(crps))
+
+    p_scores_t <- ggplot(scores_single_loc_date) +
+      geom_line(aes(x = date, y = crps, color = model),
+        show.legend = FALSE
+      ) +
+      get_plot_theme(
+        x_axis_dates = TRUE
+      ) +
+      scale_x_date(
+        date_breaks = "1 week",
+        labels = scales::date_format("%Y-%m-%d"),
+      ) +
+      xlab("") +
+      ylab("CRPS") +
+      geom_vline(aes(xintercept = forecast_date), linetype = "dashed") +
+      scale_color_manual(values = colors$model_colors)
+
+    p_scores_avg <- ggplot(scores_avg) +
+      geom_bar(aes(x = model, y = avg_crps, fill = model),
+        stat = "identity", position = "dodge",
+        show.legend = FALSE
+      ) +
+      get_plot_theme(
+        x_axis_dates = FALSE
+      ) +
+      xlab("Model") +
+      ylab("Mean CRPS") +
+      scale_fill_manual(values = colors$model_colors)
+
+    p_forecasts <- ggplot(quantiles_wide_forecast) +
+      geom_point(aes(x = date, y = eval_data),
+        fill = "white", size = 1, shape = 21,
+        show.legend = FALSE
+      ) +
+      geom_point(
+        aes(x = date, y = calib_data),
+        color = "black", show.legend = FALSE
+      ) +
+      geom_line(
+        aes(
+          x = date, y = `0.5`,
+          color = model_type
+        )
+      ) +
+      geom_ribbon(
+        aes(
+          x = date, ymin = `0.025`, ymax = `0.975`,
+          fill = model_type
+        ),
+        alpha = 0.1
+      ) +
+      geom_ribbon(
+        aes(
+          x = date, ymin = `0.25`, ymax = `0.75`,
+          fill = model_type
+        ),
+        alpha = 0.2,
+      ) +
```

Comment:
use lineribbon from ggdist?
since each geom uses `x = date`, you can specify `x=date` in the original `ggplot` call and omit it from all the `geom` calls.

Reply:
I'm going to push back on this here, I just prefer using `geom_ribbon` explicitly as I have had issues using `lineribbon` with multiple colors and facets. Not saying it wasn't a user error but I prefer to just be verbose and have control than not be able to add all the aesthetic layers I want properly.

Reply:
I would approve without this change.